src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -463,6 +470,140 @@ PRIMARY KEY ({primaryKeys})
             }
         }
 
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+        }
+
+        public async Task<SqlTriggerMetrics> GetMetricsAsync()
+        {
+            Debug.Assert(!(this._changeMonitor is null));
+
+            return new SqlTriggerMetrics
+            {
+                UnprocessedChangeCount = await this._changeMonitor.GetUnprocessedChangeCountAsync(),
```

Comment:
Wrap the call to get the change count in a try/catch and track error event if it fails.

Reply:
The method body of `GetUnprocessedChangeCountAsync` is entirely enclosed within a try-block. I am not sure if it will make sense to have extra logging here.

Reply:
Ah right, ok yeah that's enough then.